js/navigation/index.js

```diff
@@ -1,82 +1,73 @@
 // @flow
 
 import React, { Component } from 'react'
+import { BackHandler } from 'react-native'
 import { connect } from 'react-redux'
-import { StackNavigator, addNavigationHelpers } from 'react-navigation'
+import { addNavigationHelpers, NavigationActions } from 'react-navigation'
+import { createReduxBoundAddListener } from 'react-navigation-redux-helpers'
 
 import type { MapStateToProps } from 'react-redux'
-import type { NavigationState } from 'react-navigation/src/TypeDefinition'
+import type { NavigationState } from 'react-navigation'
 
-import ROUTES from './routes'
-import LoginScreen from '../components/LoginScreen'
-import ProfileScreen from '../components/ProfileScreen'
+import MainNavigator from './MainNavigator'
 
-import type { ApiUserProfile } from '../utils/api'
 import type { Dispatch } from '../actions/types'
 
-// Initialize the stack navigator
-const HomeRoutes = {
-  [ROUTES.HOME]: {
-    screen: LoginScreen
-  },
-
-  [ROUTES.PROFILE]: {
-    screen: ProfileScreen
-  }
-}
-
-export const HomeNavigator = StackNavigator(
-  {
-    ...HomeRoutes
-  },
-  {
-    initialRouteName: ROUTES.HOME,
-
-    // Let each screen handle the header and navigation
-    headerMode: 'none'
-  }
-)
-
-export const ProfileNavigator = StackNavigator(
-  {
-    ...HomeRoutes
-  },
-  {
-    initialRouteName: ROUTES.PROFILE,
-
-    // Let each screen handle the header and navigation
-    headerMode: 'none'
-  }
-)
+/**
+ * A listener of the new react-navigation redux middleware.
+ * The parameter must be the same used in createReactNavigationReduxMiddleware function.
```

Comment:
to make sure the same value is used, I suggest to define a `const` referenced by both functions

Reply:
Ok i'll add it.